models/migrations/migrations_test.go

```diff
@@ -85,21 +86,11 @@ func removeAllWithRetry(dir string) error {
 	return err
 }
 
-// newEngine sets the xorm.Engine
-func newEngine() (*xorm.Engine, error) {
-	x, err := db.NewEngine()
-	if err != nil {
-		return x, fmt.Errorf("Failed to connect to database: %v", err)
+func newXORMEngine() (*xorm.Engine, error) {
+	if err := db.InitEngine(context.Background()); err != nil {
+		return nil, err
 	}
-
-	x.SetMapper(names.GonicMapper{})
-	// WARNING: for serv command, MUST remove the output to os.stdout,
-	// so use log file to instead print to stdout.
-	x.SetLogger(db.NewXORMLogger(setting.Database.LogSQL))
-	x.ShowSQL(setting.Database.LogSQL)
-	x.SetMaxOpenConns(setting.Database.MaxOpenConns)
-	x.SetMaxIdleConns(setting.Database.MaxIdleConns)
-	x.SetConnMaxLifetime(setting.Database.ConnMaxLifetime)
+	x := unittest.GetXORMEngine()
```

Comment:
Looks like we initialize a Xorm Engine in `db` package, but get it from `unittest` package?

Reply:
I think you want to keep xorm.Engine away from non-test code.

Do you want to expose xorm.Engine from db package directly?

Reply:
Hm, I don't know. Maybe! But I concern if unittest's engine has been initialized here.

Reply:
Let's wait and see if the CI fails 😊